tidy3d/components/autograd/derivative_utils.py

```diff
@@ -199,23 +314,22 @@ def evaluate_flds_at(
         """Compute the value of an dict with keys Ex, Ey, Ez at a set of spatial locations."""
 
         xs, ys, zs = spatial_coords.T
-        edge_index_dim = "edge_index"
+        edge_index_dim = "i"
 
         interp_kwargs = {}
         for dim, locations_dim in zip("xyz", (xs, ys, zs)):
-            # only include dims where the data has more than 1 coord, to avoid warnings and errors
-            if True or all(np.array(fld.coords).size > 1 for fld in fld_dataset.values()):
-                interp_kwargs[dim] = xr.DataArray(locations_dim, dims=edge_index_dim)
+            # filter out any infinity values to use LARGE_NUMBER and get 0 in the interp()
+            locations_dim = np.nan_to_num(locations_dim, posinf=LARGE_NUMBER, neginf=-LARGE_NUMBER)
+            interp_kwargs[dim] = xr.DataArray(locations_dim, dims=edge_index_dim)
 
         components = {}
         for fld_name, arr in fld_dataset.items():
             arr_interp = arr.interp(
-                **interp_kwargs,
-                assume_sorted=True,
-                kwargs={"bounds_error": False, "fill_value": None},
+                **interp_kwargs, assume_sorted=True, kwargs=dict(fill_value=None)
```

Comment:
I think this still needs `bounds_error=False`, because according to [xarray's docs](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.interp.html), `da.interp()` will choose the default value from either [interp1d](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp1d.html) or [interpn](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interpn.html), the latter has a default of `bounds_error=True`, so this will fail for >1d interpolation.

Reply:
hm, but the thing is that I'm doing > 1d interpolation here,  and often out of bounds, so why wouldn't it be failing?

Reply:
well it worked with `bounds_error=False` so I'll just put it in for extra security

Reply:
Ok interesting, wouldn't have expected that. But still yeah I think better to keep it explicit.